static/app/components/events/interfaces/spans/spanTreeModel.tsx

```diff
@@ -772,6 +773,10 @@ class SpanTreeModel {
             false
           );
 
+          const startTimeDelta =
+            this.span.start_timestamp - parsedRootSpan.span.start_timestamp;
+          adjustEmbeddedTransactionTimestamps(parsedRootSpan, startTimeDelta);
```

Comment:
@0Calories Can this adjustment be behind a feature flag? It might have subtle impact of how customers interpret the adjust span data without having checked the timestamps beforehand.

Reply:
@dashed I don't mind adding a feature flag, but I'm not too sure what you mean here. Does the warning icon tooltip not lessen this impact enough?

Reply:
@0Calories I think ***always*** adjusting the timestamps might be overkill; hence why I suggested the feature flag. 

I'm wondering if it's possible to measure the clock drift in percentage with respect to the parent span. Something like if `1 - parsedRootSpan.span.start_timestamp / this.span.start_timestamp` is greater than `N%`, then we apply the clock drift correction. 

My intuition is that we'd apply the clock drift correction only whenever it's visually jarring. And the threshold for when it's visually jarring is `N%`.

Reply:
@dashed I see what you mean, I think I'll take your suggestion to always adjust when `startTimeDelta > 0`, and then use the clock drift % measurement for determining whether to adjust when `startTimeDelta < 0`.

One thing to note however, is that this % measurement may not be able to capture cases when an embedded transaction starts somewhere around the middle of the parent, but has spans that end after the parent span's end timestamp, which is definitely incorrect. This for example:
![image](https://user-images.githubusercontent.com/16740047/163198196-71793c49-7dca-4d76-9849-eddc21ee315c.png)

This case may not be caught by the drift percentage threshold if the embedded transaction was shifted a bit more to the left, but it would still be visually confusing, since the embedded transaction is not actually within its parent.

Do you think maybe instead, I could check and make the adjustment if any spans in the embedded transaction end after the parent? The only downside is that this would require another loop through the entire embedded transaction

Reply:
> Do you think maybe instead, I could check and make the adjustment if any spans in the embedded transaction end after the parent?

I think that could work. 

> The only downside is that this would require another loop through the entire embedded transaction

I think there is opportunity to optimize this such that we visit each span at most once. This could come later once the clock drift feature has settled.